internal/services/middleware.go

```diff
@@ -19,7 +20,13 @@ func RecoveryMiddleware(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		defer func() {
 			if err := recover(); err != nil {
-				ctxlogrus.Extract(r.Context()).
+				logger := ctxlogrus.Extract(r.Context())
+
+				if logger.Logger.Out == ioutil.Discard {
```

Comment:
not sold on this check, but atm that's the only way to detect the null logger. 

https://github.com/grpc-ecosystem/go-grpc-middleware/issues/467

Is there a way to tell if its an API hit or not?

Reply:
Maybe this?
```go
isGrpc := (&grpcweb.WrappedGrpcServer{}).IsGrpcWebRequest(r)
if isGrpc {
	logger = ctxlogrus.Extract(r.Context())
} else {
	logger = traces.Logger(r.Context())
}
```
https://pkg.go.dev/github.com/improbable-eng/grpc-web/go/grpcweb@v0.15.0#WrappedGrpcServer.IsGrpcWebRequest
Still ugly, but maybe a bit better.